libs/state/effects/src/lib/rx-effects.ts

```diff
@@ -3,24 +3,43 @@ import {
   DestroyRef,
   ErrorHandler,
   inject,
+  Injector,
+  runInInjectionContext,
 } from '@angular/core';
 import { from, Subscription } from 'rxjs';
 import { SideEffectFnOrObserver, SideEffectObservable } from './types';
 
 interface RxEffects {
   register<T>(
     observable: SideEffectObservable<T>,
-    sideEffectOrObserver?: SideEffectFnOrObserver<T>
+    sideEffectOrObserver?: SideEffectFnOrObserver<T>,
   ): Fn;
   onDestroy: (fn: Fn) => Fn;
 }
 
 type Fn = () => void;
 
 export type RxEffectsSetupFn = (
-  cfg: Pick<RxEffects, 'register' | 'onDestroy'>
+  cfg: Pick<RxEffects, 'register' | 'onDestroy'>,
 ) => void;
 
+export type RxEffectsOptions = {
+  injector?: Injector;
+};
+
+function getInjectorFromOptions<
```

Comment:
Hint to reviewer: 
I duplicated this function otherwise I would have e.g. exported it from `@rx-angular/state` in order to use it in both places which results in a public API-surface. 

Probably no user will ever try to import this function but I did not like it.

Wdyt?

Reply:
Since state depends on cdk we could move it in `cdk/internals/core`  secondary entry point. However I'm also not sure about the need of de-duplicating this fn, it is quite specific and it feels overkill to factor this.